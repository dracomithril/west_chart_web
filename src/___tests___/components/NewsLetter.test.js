/**
 * Created by XKTR67 on 10/21/2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import NewsLetter from "../../components/WestLetter";

const initial_state = require('../data/initial_state.json');
const westletters_state = require('../data/WestLetter.json');

Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureMockStore([]);
describe('<WestLetter/>', () => {
  it('renders without crashing ChartPresenter', () => {
    const store = mockStore(initial_state);
    const wrapper = shallow(
      <NewsLetter data={[]} />, {
        context: { store },
        childContextTypes: { store: PropTypes.object },
      },
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });
  it('renders users', () => {
    const store = mockStore(initial_state);
    const wrapper = shallow(
      <NewsLetter data={westletters_state} />, {
        context: { store },
        childContextTypes: { store: PropTypes.object },
      },
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

});
