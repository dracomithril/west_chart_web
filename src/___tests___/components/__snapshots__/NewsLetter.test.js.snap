// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WestLetter/> renders users 1`] = `
<div>
  <div
    className="beta"
  >
    <h1>
      This site is still in BeTa version
    </h1>
    <h2>
      Soon we will be able to track WestLetters in each week ;)
    </h2>
  </div>
  <h4>
    We have 
    7
     week of 
    2018
  </h4>
  <div
    style={
      Object {
        "display": "inline-flex",
      }
    }
  >
    <div
      key="1707149242852457_1938133309754048"
      style={
        Object {
          "border": "1px black solid",
          "display": "block",
          "padding": 2,
        }
      }
    >
      <input
        type="checkbox"
      />
      <span
        hidden={true}
      >
        1707149242852457_1938133309754048
      </span>
      <OverlayTrigger
        defaultOverlayShown={false}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="tt_1707149242852457_1938133309754048"
            placement="right"
          />
        }
        placement="bottom"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <Image
          bsClass="img"
          circle={false}
          responsive={false}
          rounded={false}
          src="https://graph.facebook.com/v2.10/10202921980928649/picture?height=50"
          thumbnail={false}
        />
      </OverlayTrigger>
      <div>
        <span>
          2017-2-24
        </span>
        <br />
        <span>
          week:
          8
        </span>
        <br />
        <span>
          is added
        </span>
      </div>
    </div>
    <div
      key="1707149242852457_1941491366084909"
      style={
        Object {
          "border": "1px black solid",
          "display": "block",
          "padding": 2,
        }
      }
    >
      <input
        type="checkbox"
      />
      <span
        hidden={true}
      >
        1707149242852457_1941491366084909
      </span>
      <OverlayTrigger
        defaultOverlayShown={false}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="tt_1707149242852457_1941491366084909"
            placement="right"
          />
        }
        placement="bottom"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <Image
          bsClass="img"
          circle={false}
          responsive={false}
          rounded={false}
          src="https://graph.facebook.com/v2.10/10202921980928649/picture?height=50"
          thumbnail={false}
        />
      </OverlayTrigger>
      <div>
        <span>
          2017-3-3
        </span>
        <br />
        <span>
          week:
          9
        </span>
        <br />
        <span>
          is added
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<WestLetter/> renders without crashing ChartPresenter 1`] = `
<div>
  <div
    className="beta"
  >
    <h1>
      This site is still in BeTa version
    </h1>
    <h2>
      Soon we will be able to track WestLetters in each week ;)
    </h2>
  </div>
  <h4>
    We have 
    7
     week of 
    2018
  </h4>
  <div
    style={
      Object {
        "display": "inline-flex",
      }
    }
  />
</div>
`;
